enos/modules/aws_client_windows/main.tf

```diff
@@ -0,0 +1,347 @@
+                  $ImdsHeaders = @{'X-aws-ec2-metadata-token' = $ImdsToken}
+                  $AuthorizedKey = (Invoke-WebRequest -Uri 'http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key' -Headers $ImdsHeaders -UseBasicParsing).Content
+                  $AuthorizedKeysPath = 'C:\ProgramData\ssh\administrators_authorized_keys'
+                  New-Item -Path $AuthorizedKeysPath -ItemType File -Value $AuthorizedKey -Force
+
+                  ## Ensure the SSH agent pulls in the new key.
+                  Set-Service -Name ssh-agent -StartupType "Automatic"
+                  Restart-Service -Name ssh-agent
+
+                  # Create a non-admin user to be used for RDP connection. This
+                  # is needed since the scheduled task that runs pyautogui
+                  # doesn't work in an Administrator context.
+                  ## Create a local user
+                  $Username = "${local.test_username}"
+                  $Password = ConvertTo-SecureString "${local.test_password}" -AsPlainText -Force
+                  New-LocalUser $Username -Password $Password -FullName "Auto Login User" -Description "User for Auto Login"
+                  Add-LocalGroupMember -Group "Administrators" -Member $Username
+
+                  # Set registry keys for auto-login
+                  $regPath = "HKLM:\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon"
+                  Set-ItemProperty -Path $regPath -Name "AutoAdminLogon" -Value "1" -Type String
+                  Set-ItemProperty -Path $regPath -Name "DefaultUsername" -Value $Username -Type String
+                  Set-ItemProperty -Path $regPath -Name "DefaultPassword" -Value "${local.test_password}" -Type String
+                  Set-ItemProperty -Path $regPath -Name "DefaultDomainName" -Value "$env:COMPUTERNAME" -Type String
+
+                  # Set all active networks to Private, which disables the
+                  # "allow PC to be discoverable" notification. Otherwise, the
+                  # notification will interfere with the automation)
+                  # todo: None of these worked... ended up needing to start the
+                  # python script with an Esc to dismiss the notification
+                  Get-NetConnectionProfile | Where-Object {$_.NetworkCategory -ne "Private"} | Set-NetConnectionProfile -NetworkCategory Private
+
+                  netsh advfirewall firewall set rule group="Network Discovery" new enable=No
+
+                  New-ItemProperty -Path "HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Network" -Name "NewNetworkWindowOff" -PropertyType DWord -Value 1 -Force
+                  New-ItemProperty -Path "HKLM:\\System\\CurrentControlSet\\Control\\Network" -Name "NewNetworkWindowOff" -PropertyType DWord -Value 1 -Force
+                  Restart-Service -Name NlaSvc -Force
```

Comment:
Not sure if we need to keep this. I was trying to figure out how to dismiss that notification that appears when you first start up, but I just ended up having the testing script start by hitting "Esc" which either dismisses the notification (if it's there) or does nothing.